src/ripple/app/main/GRPCServer.h

```diff
@@ -234,20 +235,26 @@ class GRPCServerImpl final
 
 };  // GRPCServerImpl
 
-class GRPCServer
+class GRPCServer : public Stoppable
 {
 public:
-    explicit GRPCServer(Application& app) : impl_(app){};
+    explicit GRPCServer(Application& app, Stoppable& parent)
+        : Stoppable("GRPCServer", parent), impl_(app)
+    {
+    }
 
     GRPCServer(const GRPCServer&) = delete;
 
     GRPCServer&
     operator=(const GRPCServer&) = delete;
 
     void
-    run();
+    onStart() override;
+
+    void
+    onStop() override;
 
-    ~GRPCServer();
+    ~GRPCServer() = default;
```

Comment:
Interesting note: the core guidelines used to advise against adding `override` to destructors. They've since changed the advice to adding it. My own feeling: it's OK to add it, but it clearly isn't as useful as for regular functions (there's no danger of introducing an overload by mistake, for example), but it still tags it as a virtual function. So probably good to add to new code, but I'm not in a hurry to fix existing code.